### ISSUE TITLE
False positive `:unused-private-var` warning for `deftype ^:private`

### ISSUE THREAD

**Issue by Valdermeyder** [clj-kondo/clj-kondo#1222]
**version**

v2021.03.03

**problem**

False positive `:unused-private-var` warning for `deftype ^:private` event it used inside the namespace

**repro**
```
(deftype ^:private SessionStore
  [session-service])

(defn db-store [this]
  (SessionStore. this))
```
Error "warning: Unused private var user/->SessionStore"

** configuration workaround **
:unused-private-var {:exclude [user/->SessionStore]}

**expected behavior**

No warning message
